baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgumentTest.java

```diff
@@ -716,6 +716,120 @@ public void testStringConcatAssignmentSafety() {
                 .doTest();
     }
 
+    @Test
+    public void testStringFormatSafety() {
+        helper().addSourceLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "class Test {",
+                        "  void f(",
+                        "      @Safe String safeParam,",
+                        "      @Unsafe String unsafeParam,",
+                        "      @DoNotLog String doNotLogParam) {",
+                        "    String foo = String.format(\"%s\", safeParam);",
+                        "    fun(foo);",
+                        "    foo = String.format(\"%s\", unsafeParam);",
+                        "    // BUG: Diagnostic contains: Dangerous argument value: arg is 'UNSAFE'",
+                        "    fun(foo);",
+                        "    foo = String.format(\"%s\", doNotLogParam);",
+                        "    // BUG: Diagnostic contains: Dangerous argument value: arg is 'DO_NOT_LOG'",
+                        "    fun(foo);",
+                        "  }",
+                        "  private static void fun(@Safe Object obj) {}",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    public void testStringFormatWithImplicitToStringSafety() {
+        helper().addSourceLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "class Test {",
+                        "  class SafeToStringClass {",
+                        "    @Safe",
+                        "    @Override",
+                        "    public String toString() {",
+                        "      return \"safe\";",
+                        "    }",
+                        "  }",
+                        "  class UnsafeToStringClass {",
+                        "    @Unsafe",
+                        "    @Override",
+                        "    public String toString() {",
+                        "      return \"unsafe\";",
+                        "    }",
```

Comment:
The conjure generator for unsafe aliases generates code like this. Example type:

```
      ObjectTypeId:
        alias: string
        safety: unsafe
```

Reply:
I don't think this is true -- the alias generator aplies an `@Unsafe` annotation to both toString and the class. Similarly, the SafeLoggingPropagation error-prone check requires us to propagate safety information from `toString()` to the top level class.

Reply:
Ah true.  Please disregard -- I think the error I'm hunting for is elsehwere